src/FFMpeg/Media/Video.php

```diff
@@ -129,6 +129,56 @@ public function save(FormatInterface $format, $outputPathfile)
             }
         }
 
+        // Merge Filters into one command
+        $videoFilterVars = $videoFilterProcesses = [];
+        for($i=0;$i<count($commands);$i++) {
+            $command = $commands[$i];
+            if ( $command == '-vf' ) {
+                $commandSplits = explode(";", $commands[$i + 1]);
+                if ( count($commandSplits) == 1 ) {
+                    $commandSplit = $commandSplits[0];
+                    $command = trim($commandSplit);
+                    if ( preg_match("/^\[in\](.*?)\[out\]$/is", $command, $match) ) {
+                        $videoFilterProcesses[] = $match[1];
+                    } else {
+                        $videoFilterProcesses[] = $command;   
+                    }
+                } else {
+                    foreach($commandSplits as $commandSplit) {
+                        $command = trim($commandSplit);
+                        if ( preg_match("/^\[[^\]]+\](.*?)\[[^\]]+\]$/is", $command, $match) ) {
+                            $videoFilterProcesses[] = $match[1];
+                        } else {
+                            $videoFilterVars[] = $command;
+                        }                
+                    }
+                }
+                unset($commands[$i]);
+                unset($commands[$i + 1]);
+                $i++;
+            }
+        }
+        $videoFilterCommands = $videoFilterVars;
+        $lastInput = 'in';
+        foreach($videoFilterProcesses as $i => $process) {
+            $command = '[' . $lastInput .']';
+            $command .= $process;
+            $lastInput = 'p' . $i;
+            if ( $i == count($videoFilterProcesses) - 1 ) {
+                $command .= '[out]';
+            } else {
+                $command .= '[' . $lastInput . ']';
+            }
+            
+            $videoFilterCommands[] = $command;
+        }
+        $videoFilterCommand = implode(";", $videoFilterCommands);
+        
+        if ( $videoFilterCommand ) {
+            $commands[] = '-vf';
+            $commands[] = $videoFilterCommand;
+        }
```

Comment:
but merging the filters into one is not an easy job. because the structure of the class is now returning the command directly. that mean mutiple vf command is returned. we have to find a way to merge then instead to reconstruct the whole class which is a huge change of the class. this is the best i think without any problem because all commands are returned by the classes, it is easy for us to know the format and merge them into one.